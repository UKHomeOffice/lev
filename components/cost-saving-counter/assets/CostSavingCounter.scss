.ho-lev-cost-saving-counter {
  flex: 1 1 0;
  text-align: center;
  border: 8px solid #9325b2;
  border-radius: 7px;
  margin-bottom: 1vh;

  & > * {
    color: #9325b2;
    max-width: 100%;
  }
  &__heading {
    font-size: 9vh;
  }
}
