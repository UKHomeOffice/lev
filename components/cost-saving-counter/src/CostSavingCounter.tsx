import { FC, createElement as h } from 'react';
import { StandardProps, classBuilder } from '@not-govuk/component-helpers';

import '../assets/CostSavingCounter.scss';

export type CostSavingCounterProps = StandardProps & {
  /** Description for the 'costsaving' prop */
  costsaving: number
  /** Description for the 'group' prop */
  group: string
};

export const CostSavingCounter: FC<CostSavingCounterProps> = ({ costsaving, group, classBlock, classModifiers, className, ...attrs }) => {
  const classes = classBuilder('ho-lev-cost-saving-counter', classBlock, classModifiers, className);
  const title = costsaving ? `Â£${Number(costsaving).toLocaleString()}` : 'Cost saving counter';

  return (
    <div {...attrs} className={classes()}>
      <h2 className={classes('heading')}>{title}</h2>
      <p>cost saving to {group}</p>
    </div>
  );
}

export default CostSavingCounter;
