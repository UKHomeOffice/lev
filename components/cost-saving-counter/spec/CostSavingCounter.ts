import { createElement as h } from 'react';
import { mount } from 'enzyme';
import CostSavingCounter from '../src/CostSavingCounter';
import toJson from 'enzyme-to-json';

describe('CostSavingCounter', () => {
  describe('when given valid props', () => {
    const group = 'the public';
    const component = mount(h(CostSavingCounter, { costsaving: 2345.56, group: group }));

    it('renders cost saving', () => expect(component.find('h2').text()).toEqual('Â£2,345.56'));
    it('renders the group', ()=> expect(component.find('p').text()).toContain(group));
    it('contains no regressions', () => expect(toJson(component)).toMatchSnapshot());
  });
});
