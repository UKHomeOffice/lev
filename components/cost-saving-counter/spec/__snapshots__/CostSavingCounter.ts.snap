// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CostSavingCounter when given valid props contains no regressions 1`] = `
<Component
  costsaving={2345.56}
  group="the public"
>
  <div
    className="ho-lev-cost-saving-counter"
  >
    <h2
      className="ho-lev-cost-saving-counter__heading"
    >
      Â£2,345.56
    </h2>
    <p>
      cost saving to 
      the public
    </p>
  </div>
</Component>
`;
